packages/utils/crates/utils-misc/src/async_iterable.rs

```diff
@@ -0,0 +1,97 @@
+pub mod wasm {
+    use futures::stream::{Stream, StreamExt};
+    use js_sys::Uint8Array;
+    use serde::Serialize;
+    use serde_wasm_bindgen;
+    use wasm_bindgen::prelude::*;
+
+    #[wasm_bindgen]
+    #[derive(Serialize, Clone)]
```

Comment:
Be careful when using `#[wasm_bindgen]` on stuff that you also serialize/deserialize using `serde_wasm_bindgen`. The definition that gets generated via `wasm_bindgen` is not always compatible with what you serialize - for example, enums are not getting deserialized properly and make the code panic. This is volatile stuff.

Reply:
Haven't had any issues so far ... but you need to add `#[wasm_bindgen]` in front of any `enum`. Stuff like 
```
enum SomeComplexEnum {
   Struct { a: u32 },
   Foo(u8)
}
```
is certainly not working but simple things such as
```
enum Message {
  OK,
  NOT_OK,
}
```
has not been an issue